### ISSUE TITLE
"no rust sdk defined for this project" when trying to run cargo

### ISSUE THREAD

**Issue by camjackson** [Vektah/idea-rust#28]
After installing the plugin, I created a new Rust project. I have the project set to use Rust 0.13.0-nightly, and the module using the project's SDK. I then create a cargo run config, and try to run it. I always get the above error.

Here are some screenshots of my configs and the result:

![screen shot 2015-01-05 at 11 33 02 pm](https://cloud.githubusercontent.com/assets/1930451/5613163/f7401a54-9533-11e4-8c43-fb8c5bb2596e.png)
![screen shot 2015-01-05 at 11 33 59 pm](https://cloud.githubusercontent.com/assets/1930451/5613165/f7421e8a-9533-11e4-9add-6e5f7f9686d4.png)
![screen shot 2015-01-05 at 11 34 48 pm](https://cloud.githubusercontent.com/assets/1930451/5613166/f74559d8-9533-11e4-98f1-11ff1451cc9b.png)
![screen shot 2015-01-05 at 11 35 28 pm](https://cloud.githubusercontent.com/assets/1930451/5613164/f740438a-9533-11e4-90cf-579baf082266.png)
![screen shot 2015-01-05 at 11 36 43 pm](https://cloud.githubusercontent.com/assets/1930451/5613162/f73cfdba-9533-11e4-9eb8-5a6459dbf583.png)

**Comment by camjackson** [comment]
And yes, I do have cargo and rustc at `/usr/local/bin`.

**Comment by hastebrot** [comment]
Same here on Windows 7, IntelliJ 14 and Cargo installed.

**Comment by Vektah** [comment]
I'm having some difficulty reproducing this.

Could you try restarting intellij and re-opening the project? I can only guess the sdk is getting cached somewhere...

**Comment by camjackson** [comment]
Restarting seems to have helped, although now I get a different message: 

>Error running run: Cannot run program "cargo" (in directory "/Users/cjacks/code/rust-learning"): error=2, No such file or directory

I tried changing the working directory of the run configuration to `/usr/local/bin` and that didn't help.

**Comment by Vektah** [comment]
Currently cargo is assumed to be on your path. Can you run `cargo` from the command line?

The working directory needs to be your project directory otherwise cargo will be unable to find the toml file.

**Comment by camjackson** [comment]
Yeah I figured as much for both of those things. Have no trouble running
cargo or rustc from the command line, and the working directory is the
project dir.

**Comment by Blackhole16** [comment]
I have the same configuration as @camjackson. I have rustc in my /usr/share/bin but the plugin says I don't have an SDK defined for the project.

**Comment by camjackson** [comment]
@Blackhole16 Did you try restarting your IDE? That got rid of the 'no SDK' problem for me.

I'm still getting the same error I mentioned earlier though - `...no such file...`.

**Comment by Blackhole16** [comment]
I restarted the IDE several times and even reinstalled it completely. Nothing changed. It also showed all reserrved words like they are wrong (red underlined), eg. loop, for, while etc. That's why I switched to sublime, which actually has an amazing rust integration. Maybe I'll come back later when this plugin also has auto-completion :)

**Comment by hastebrot** [comment]
Hmm, I can't reproduce this issue anymore. Cargo run works for me on both version `0.0.4` and `master`.

**Comment by hastebrot** [comment]
However, when I reconfigure the Rust SDK I get the error "No Rust sdk defined for this project".

**Comment by hastebrot** [comment]
What worked without errors:

- File > Project Structure...
  - Platform Settings > SDKs
    - Add New SDK > Rust SDK.
    - On Windows select `C:\Rust\bin`. OK.
  - Project Settings > Project
    - Project SDK: `Rust 1.0.0-nightly`
  - Project Settings > Modules
    - Dependencies > Module SDK: `Rust 1.0.0-nightly`
- Run > Edit Configurations...
  - Add New Configuration > Cargo > OK.
- Restart IntelliJ.
- Run > Run 'Unnamed'

I can reproduce the issue by removing the Rust SDK and follow these steps without restarting IntelliJ.

**Comment by camjackson** [comment]
Closing this issue, as restarting the IDE fixes it. I've opened #37 instead.

**Comment by Vektah** [comment]
I'm going to re-open this as it is still a bug, even if there is a workaround.

**Comment by ivstas** [comment]
For me, the solution to solve the issue was to **explicitly** set *module* sdk to **Rust 1.0.0-nightly** rather than use **project sdk**. And restarting IntelliJ after that.

**Comment by Rodney-Gabriel** [comment]
guys, I really need your help. I'm just getting started in programming and in Rust and Im trying to get the SDK configured to look like the one above with the nightly build. How do i do that? I just want to learn to program and getting things set up seems like a pain. Please help.

**Comment by camjackson** [comment]
@Rodney-Gabriel You might be better off trying a different editor, at least for now, as development on this plugin seems to have stalled. Maybe try Atom or Sublime? (I personally am using Vim with [rust.vim](https://github.com/rust-lang/rust.vim), but if you're just starting with coding, I wouldn't recommend trying to learn Vim at the same time.)

**Comment by Rodney-Gabriel** [comment]
@camjackson Thank you. I have both of those editors I was under the impression that I needed a full fledged IDE to run and compile the code to test it and know you can't do that in an editor. So how would you suggest I run the code after I write it in either Sublime or Atom?

**Comment by camjackson** [comment]
You can compile and run your code from the command line using `rustc` or `cargo`. Assuming you've already installed rust on your machine (which gives you both `rustc` and `cargo`), see [here](https://doc.rust-lang.org/stable/book/hello-world.html) and then [here](https://doc.rust-lang.org/stable/book/hello-cargo.html).

If you have more questions, probably best to move the discussion to one of the forums listed under 'community' at https://www.rust-lang.org/, rather than here. Good luck!
